internal/balancergroup/balancergroup.go

```diff
@@ -403,7 +408,7 @@ func (bg *BalancerGroup) Remove(id string) {
 
 	sbToRemove, ok := bg.idToBalancerConfig[id]
 	if !ok {
-		bg.logger.Infof("balancer group: trying to remove a non-existing locality from balancer group: %v", id)
+		bg.logger.Warningf("Child policy for locality %q does not exist in the balancer group", id)
```

Comment:
Should this even be `Error` so our tests fail?  I.e. does it imply a coding error on our part?